File: app/Models/BienTheSanPham.php
<?php 
namespace App\Models;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Support\Facades\Session;
use App\Models\KichThuocSanPham;
use App\Models\MauSanPham;
use App\Models\AnhSanPham;
use App\Models\TrangThai;
use App\Models\Magiamgia;
class BienTheSanPham extends Model{
    use HasFactory;
    public $timestamps =  false;
    protected $table = 'bien_the_san_pham';
    protected $primaryKey = 'id';
    protected $fillable = ['id','ten_bien_the','san_pham_id','mau_id','kich_thuoc_id','so_luong','gia_ban','id_anh','trang_thai','ma_giam_gia','gia_nhap'];
    public function anh()
    {
        return $this->belongsTo(AnhSanPham::class, 'id_anh');
    }
     public function mau()
    {
        return $this->belongsTo(MauSanPham::class, 'mau_id');
    }
      public function kichthuoc()
    {
        return $this->belongsTo(KichThuocSanPham::class, 'kich_thuoc_id');
    }
     public function trangthai()
    {
        return $this->belongsTo(TrangThaiSanPham::class, 'trang_thai');
    }
       public function magiamgia()
    {
        return $this->belongsTo(Magiamgia::class, 'ma_giam_gia');
    }
}

File: lib/models/bien_the_san_pham.dart
import 'package:kingofshoes/models/anh_sp.dart';

class BienTheSanPham {
  final int id;
  final int san_pham_id;
  final String ten_bien_the;
  final int mau_id;
  final int kich_thuoc_id;
  int so_luong;
  final double gia_ban;
  final int id_anh;
  final String trang_thai;
  final int ma_giam_gia;
  final double? gia_nhap;
  final AnhSp? anhSp;
  final String? mo_ta;

  BienTheSanPham(
      {required this.id,
      required this.san_pham_id,
      required this.ten_bien_the,
      required this.mau_id,
      required this.kich_thuoc_id,
      required this.so_luong,
      required this.gia_ban,
      required this.id_anh,
      required this.trang_thai,
      required this.ma_giam_gia,
      required this.gia_nhap,
      required this.anhSp,
      required this.mo_ta});

  factory BienTheSanPham.fromJson(Map<String, dynamic> json) {
    return BienTheSanPham(
      id: json['id'] ?? 0,
      san_pham_id: json['san_pham_id'] ?? 0,
      ten_bien_the: json['ten_bien_the'] ?? '',
      mau_id: json['mau_id'] ?? 0,
      kich_thuoc_id: json['kich_thuoc_id'] ?? 0,
      so_luong: json['so_luong'] ?? 0,
      gia_ban: json['gia_ban'] != null ? double.parse(json['gia_ban']) : 0.0,
      id_anh: json['id_anh'] ?? 0,
      trang_thai: json['trang_thai'] ?? '',
      ma_giam_gia: json['ma_giam_gia'] ?? 0,
      gia_nhap: json['gia_nhap'] != null ? double.parse(json['gia_nhap']) : 0.0,
      anhSp: json['anh'] != null ? AnhSp.fromJson(json['anh']) : null,
      mo_ta: json['mo_ta'] ?? '',
    );
  }
}
